### PR TITLE
Libimagrt/edit

### PR DESCRIPTION
**Pull request by matthiasbeyer**
`Entry::edit()` functionality to edit the file contents.

@TheNeikos can you review?

### CODE REVIEW

File: libimagrt/src/edit.rs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    use std::io::Seek;
+    use std::io::Read;
+    use std::io::SeekFrom;
+    use std::io::Write;
+
+    let file = NamedTempFile::new();
+    if file.is_err() {
+        return Err(RuntimeError::new(RuntimeErrorKind::Instantiate, None));
+    }
+    let file      = file.unwrap();
```
Comment (TheNeikos) [inline libimagrt/src/edit.rs]: You can remove this by implementing `From<IoError> for RunTimeError`

File: libimagrt/src/edit.rs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    let file_path = file.path();
+    let file      = file.reopen();
+
+    if file.is_err() {
+        return Err(RuntimeError::new(RuntimeErrorKind::IOError, Some(Box::new(file.err().unwrap()))));
+    }
+
+    let mut file = file.unwrap();
```
Comment (TheNeikos) [inline libimagrt/src/edit.rs]: same here

File: libimagrt/src/edit.rs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    file.write_all(&s.clone().into_bytes()[..]);
+
+    if let Err(e) = file.sync_data() {
```
Comment (TheNeikos) [inline libimagrt/src/edit.rs]: same here

File: libimagrt/src/edit.rs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+        match exit_status.map(|s| s.success()) {
+            Ok(true) => {
+                file.sync_data()
+                    .and_then(|_| file.seek(SeekFrom::Start(0)))
+                    .and_then(|_| file.read_to_string(s))
+                    .map(|_| ())
+                    .map_err(|e| RuntimeError::new(RuntimeErrorKind::IOError, Some(Box::new(e))))
+            },
+            Ok(_)    => Err(RuntimeError::new(RuntimeErrorKind::ProcessExitFailure, None)),
```
Comment (TheNeikos) [inline libimagrt/src/edit.rs]: Does false not work here?

Comment (matthiasbeyer) [inline libimagrt/src/edit.rs]: Jup, should work. :astonished:

**Comment by TheNeikos** [comment]
Looking good code wise :+1:

**Pull request merged**